sites/docs/src/content/docs/tutorials/switching_barnch_to_main/switching_master_to_main.md

```diff
@@ -0,0 +1,99 @@
+---
+title: "Transitioning from `master` to `main` branches in pipelines"
+subtitle: Guide on how to switch a pipeline from the default branch `master` to `main`
+weight: 20
+---
+
+This tutorial will guide you through changing the pipeline default branch form `master` to `main`.
+
+## Prequisites
+
+- Admin rights on the GitHub repository where the change taking place
+- nf-core tools
+  - It is recommended that the version match the version of the template in your pipeline
+
+> [!WARNING]
+> Make sure you are on using same version of nf-core/tools as the template version of your branch.
+> This will avoid adding other changes not needed for the branch name switch.
+
+## Instructions
+
+1. Open your repository on GitHub (make sure you're on master)
+2. Go to the `branches` page (to the right of the `branches` drop down where it says 'master')
+
+![image](https://hackmd.io/_uploads/rkc8GXJWgx.png)
+
+3. Under the default branch section of the page, press the triple dot menu on the `master` branch row, and press 'Rename branch'
+
+> [NOTE]
+> You will need permissions to change this setting. If you don't have permission, ask the @core-team to do it for you.
+
+![image](https://hackmd.io/_uploads/Bkpdz7kWlg.png)
+
+4.  Rename `master` to `main`
+5.  Press the 'learn more' text, and copy and paste the displayed instructions somewhere safe.
+
+> [NOTE]
+> These instructions will be usefur for all pipeline contributors.
+
+![image](https://hackmd.io/_uploads/SkiIQXyWlx.png)
+
+6. Press 'Rename branch'
+7. Go back to 'Code' tab to verify you are now on `main`
+8. In your local IDE (e.g. VSCode) make sure you're on the `dev` branch
+9. Run `git pull` to ensure you have the `main` branch locally
```

Comment:
isn't it get `git pull upstream main`?

Reply:
> actually I think this step is not needed. Or did you find an error without it @jfy133 ?

I think so? Otherwise I think `git config` didn't know what it was pointing at... but I don't remember exactly.

`git pull upstream main` I guess would be correct if you're working on a fork. I did this directly on the mag repo...

Reply:
maybe `git fetch upstream` would be better if we are on a fork